blockchain/storage/leveldb/leveldb.go

```diff
@@ -25,7 +31,13 @@ func Factory(config map[string]interface{}, logger hclog.Logger) (storage.Storag
 
 // NewLevelDBStorage creates the new storage reference with leveldb
 func NewLevelDBStorage(path string, logger hclog.Logger) (storage.Storage, error) {
```

Comment:
Do we need another constructor which takes on `opt.Options` instance if someone wants to use another configuration? I am saying this because we already had a similar issue reported on github where someone wanted to use a different db, and a different set of options, but we didn't support it.

```suggestion
func NewLevelDBStorage(path string, logger hclog.Logger) (storage.Storage, error) {
      defaultOptions := &opt.Options{/*set default options*/ }
      
      return NewLevelDBStorageWithOpt(path, logger, defaultOptions)
}

func NewLevelDBStorageWithOpt(path string, logger hclog.Logger, opts opt.Options) (storage.Storage, error) {
   // do the magic in the existing constructor
}
```